### PR TITLE
Airflow

### PR DESCRIPTION
**Pull request by securims**

### CODE REVIEW

File: pipelines/airflow-home/dags/inference_dag.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        image=IMAGE,
+        cmds=[
+            "python",
+            "cli.py",
+            "inferencepool",
+            "./test.csv"
+        ],
+        task_id="run_inference",
+        in_cluster=False,
+        env_vars={"WANDB_API_KEY": "c1bc1fb8ecae6285b816e39a36672e42d0187253"},
```
Comment (truskovskiyk) [inline pipelines/airflow-home/dags/inference_dag.py]: please remove creds from this PR

File: pipelines/airflow-home/dags/train_dag.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+# The DAG object; we'll need this to instantiate a DAG
+from airflow import DAG
+
+# Operators; we need this to operate!
+from airflow.operators.bash import BashOperator
+with DAG(
+    'tutorial',
+    # These args will get passed on to each operator
+    # You can override them on a per-task basis during operator initialization
+    default_args={
```
Comment (truskovskiyk) [inline pipelines/airflow-home/dags/train_dag.py]: name of this file is `train_dag.py`, should we rename it ?

File: pipelines/airflow-home/dags/training_dag.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        volume_mounts=[volume_mount],
+    )
+
+    train_and_upload_model = KubernetesPodOperator(
+        name="train_and_upload_model",
+        image=IMAGE,
+        cmds=["python", "cli.py", "train", "./test.csv"],
+        task_id="train_and_upload_model",
+        in_cluster=False,
+        env_vars={"WANDB_API_KEY": "c1bc1fb8ecae6285b816e39a36672e42d0187253"},
```
Comment (truskovskiyk) [inline pipelines/airflow-home/dags/training_dag.py]: please remove key

File: pipelines/airflow-home/dags/training_dag.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        namespace="default",
+        volumes=[volume],
+        volume_mounts=[volume_mount],
+    )
+   
+    load_data >> train_and_upload_model
```
Comment (truskovskiyk) [inline pipelines/airflow-home/dags/training_dag.py]: what is happening with the model after? how is it used in training dag? don't see saving operation here

File: pipelines/airflow-home/dags/training_dag.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+
+from airflow import DAG
+from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
+from kubernetes.client import models as k8s
+
+volume = k8s.V1Volume(name="training-storage", persistent_volume_claim=k8s.V1PersistentVolumeClaimVolumeSource(claim_name="training-storage"),)
+volume_mount = k8s.V1VolumeMount(name="training-storage", mount_path="/", sub_path=None)
```
Comment (truskovskiyk) [inline pipelines/airflow-home/dags/training_dag.py:8]: should we clean this volume before each run?

**Review by truskovskiyk — COMMENTED** [review]
@securims 

kfp - looks great 
airflow - some comments 


Common question to both: 

how does the inference pipeline get access to the trained model?

File: inference/inference.py [hunk 6]
```diff
@@ -38,12 +38,16 @@ def train_model(x_train: np.ndarray, y_train: np.ndarray) -> ak.TextRegressor:
     try:
         model.save("model_autokeras", save_format="tf")
     except:
-        model.save("model_autokeras.h5")
+        model.save("model_autokeras.h5") 
+    
+    art = wandb.Artifact("model_autokeras", type="model")
+    art.add_dir("model_autokeras")
+    wandb.log_artifact(art)
```
Comment (securims) [inline inference/inference.py:45]: @truskovskiyk model is uploaded to wandb right after training

File: pipelines/airflow-home/dags/training_dag.py (hunk 4 above)
Comment (securims) [reply pipelines/airflow-home/dags/training_dag.py]: right now code is working in a way that after training model will be saved locally in folder generated by autokeras and uploaded to wandb, in  the beginning of inference last artifact will be downloaded from wandb

**Review by truskovskiyk — APPROVED** [review]
@securims got it, thanks for the work!

**Pull request merged**